src/code/z_sram_NES.c

```diff
@@ -10,128 +10,28 @@ void func_80147414(SramContext* sramCtx, s32 fileNum, s32 arg2);
-    { 0xC00, 0, 0, 0x40000000 },    // SCENE_20SICHITAI
-    { 0x102, 0, 0, 0x400 },         // SCENE_21MITURINMAE
-    { 0, 0, 0, 2 },                 // SCENE_LAST_ZORA
-    { 0, 0, 0, 0x40000000 },        // SCENE_11GORONNOSATO2
-    { 0x70, 0, 0, 0 },              // SCENE_SEA
-    { 0, 0, 0, 0 },                 // SCENE_35TAKI
-    { 0, 0, 0, 0 },                 // SCENE_REDEAD
-    { 0, 0, 0, 0x40000000 },        // SCENE_BANDROOM
-    { 0, 0, 0, 0x40000000 },        // SCENE_11GORONNOSATO
-    { 0, 0, 0, 0 },                 // SCENE_GORON_HAKA
-    { 0, 0, 0, 0 },                 // SCENE_SECOM
-    { 0x100000, 0, 0, 0x80000000 }, // SCENE_10YUKIYAMANOMURA
-    { 0, 0, 0, 0 },                 // SCENE_TOUGITES
-    { 0, 0, 0, 0 },                 // SCENE_DANPEI
-    { 0, 0, 0, 0 },                 // SCENE_IKANAMAE
-    { 0, 0, 0, 0 },                 // SCENE_DOUJOU
-    { 0, 0, 0, 0 },                 // SCENE_MUSICHOUSE
-    { 0, 0, 0, 0 },                 // SCENE_IKNINSIDE
-    { 0, 0, 0, 0 },                 // SCENE_MAP_SHOP
-    { 0x400, 0, 0, 0 },             // SCENE_F40
-    { 0x400, 0, 0, 0 },             // SCENE_F41
-    { 0x100000, 0, 0, 0x80000000 }, // SCENE_10YUKIYAMANOMURA2
-    { 0, 0, 0, 0x100 },             // SCENE_14YUKIDAMANOMITI
-    { 0x400, 0, 0, 0 },             // SCENE_12HAKUGINMAE
-    { 0, 0, 0, 0x80 },              // SCENE_17SETUGEN
-    { 0, 0, 0, 0x80 },              // SCENE_17SETUGEN2
-    { 0, 0, 0, 0x80000000 },        // SCENE_SEA_BS
-    { 0, 0, 0, 0x400 },             // SCENE_RANDOM
-    { 0, 0, 0, 0 },                 // SCENE_YADOYA
-    { 0, 0, 0, 0 },                 // SCENE_KONPEKI_ENT
-    { 1, 0, 0, 0 },                 // SCENE_INSIDETOWER
-    { 0, 0, 0, 0 },                 // SCENE_26SARUNOMORI
-    { 0, 0, 0, 0 },                 // SCENE_LOST_WOODS
-    { 0, 0, 0, 2 },                 // SCENE_LAST_LINK
-    { 0, 0, 0, 0 },                 // SCENE_SOUGEN
-    { 0, 0, 0, 0 },                 // SCENE_BOMYA
-    { 0, 0, 0, 0 },                 // SCENE_KYOJINNOMA
-    { 0, 0, 0, 0 },                 // SCENE_KOEPONARACE
-    { 0, 0, 0, 0 },                 // SCENE_GORONRACE
-    { 1, 0, 0, 0 },                 // SCENE_TOWN
-    { 0, 0, 0, 0 },                 // SCENE_ICHIBA
-    { 0, 0, 0, 0x400 },             // SCENE_BACKTOWN
-    { 0x100000, 0, 0, 0x400 },      // SCENE_CLOCKTOWER
-    { 0, 0, 1, 0 },                 // SCENE_ALLEY
+typedef struct PersistentCycleFlags {
+    /* 0x0 */ u32 switch0;
+    /* 0x4 */ u32 switch1;
+    /* 0x8 */ u32 chest;
+    /* 0xC */ u32 collectible;
+} PersistentCycleFlags; // size = 0x10
+
+#define DEFINE_SCENE(_name, _enumValue, _textId, _drawConfig, _restrictionFlags, switch0, switch1, chest, collectible) \
+    { switch0, switch1, chest, collectible },
+
+#define DEFINE_SCENE_UNSET(_enumValue) { 0, 0, 0, 0 },
+
+/**
+ * Array of bitwise flags which won't be turned off on a cycle reset (will persist between cycles)
```

Comment:
Would be nice to have this descriptive comment where the bits are set in the scene table too